File: udemy-frontend/src/type/pagination.ts
export interface IPagination{
    page : number,
    size : number,
    active? : boolean,
    keyword? : string
}
export interface IPaginationResponse<T>{
    elements : T[],
    meta : IMetaResponse
}
export interface IMetaResponse{
    currentPage : number,
    pageSize : number,
    elementTotals : number,
    pageTotals : number,
}

File: udemy-backend/src/main/java/com/jwhisper/udemy/dto/Pagination.java
package com.jwhisper.udemy.dto;

import java.util.List;

import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;

@Data
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Pagination<T> {
    List<T> elements;
    Meta meta;
    @Data
    @FieldDefaults(level = AccessLevel.PRIVATE)
    public static class Meta  {
        int currentPage;
        int pageSize;
        long elementTotals;
        int pageTotals;
    }
}
